### PR TITLE
Add CliCommand wrapper that adds help URL

### PR DESCRIPTION
**Pull request by zivkan**
# Bug

Fixes: https://github.com/dotnet/sdk/pull/44614#discussion_r1827170176

## Description

The `dotnet` CLI has a command `dotnet help <command>` which will open the default browser and take you to the help page for the `<command>`.  For this to work, the commands need to tell the `dotnet` CLI what the help URL is.

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [x] Added tests
- [x] ~Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.~

### CODE REVIEW

**Comment by zivkan** [comment]
fyi @baronfel

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/DocumentedCommand.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.CommandLine;
+
+namespace NuGet.CommandLine.XPlat.Commands
+{
+    public class DocumentedCommand : CliCommand
+    {
+        public DocumentedCommand(string name, string description, string helpUrl)
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/DocumentedCommand.cs:10]: `CliCommand` makes the description optional. My gut feeling is that we should add a description to every command, so I thought it'd be fine to make it non-optional for this wrapper class. Well, I guess you could just explicitly pass in a null.

But if anyone thinks that we should follow the `CliCommand` pattern and make `string description = null`, then I should re-order these parameters.

Any thoughts?

**Review by jeffkl — APPROVED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/DocumentedCommand.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/DocumentedCommand.cs:10]: I don't think we'd ever want to ship a command without description, so I agree with just making it non-optional here. 

I'm fine with us doing a null check.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigCommand.cs [hunk 2]
```diff
@@ -95,37 +96,37 @@ internal static void Register(CommandLineApplication app)
...
         {
-            var ConfigCmd = new CliCommand(name: "config", description: Strings.Config_Description);
+            var ConfigCmd = new DocumentedCommand(name: "config", description: Strings.Config_Description, "https://aka.ms/dotnet/nuget/config");
             ConfigCmd.Options.Add(HelpOption);
 
             // Options directly under the verb 'config'
 
             // noun sub-command: config paths
-            var PathsCmd = new CliCommand(name: "paths", description: Strings.ConfigPathsCommandDescription);
+            var PathsCmd = new DocumentedCommand(name: "paths", description: Strings.ConfigPathsCommandDescription, "https://aka.ms/dotnet/nuget/config/paths");
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigCommand.cs:105]: did these aka.ms links exist already?

**Review by nkolev92 — APPROVED** [review]

**Review by martinrrm — APPROVED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigCommand.cs (hunk 2 above)
Comment (zivkan) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigCommand.cs:105]: yes, I created them as I was making the code changes.

**Review by Nigusu-Allehu — APPROVED** [review]

**Pull request merged**